src/analytics.ts

```diff
@@ -19,12 +19,33 @@ export function createAnalyticsInstance(options?: Options) {
         event_cache: Array<{ event: keyof TEvents; payload: TEvents[keyof TEvents] }> = [],
         page_view_cache: Array<{ current_page: string; platform: string; user_id: string }> = []
 
-    const initialise = ({ growthbookKey, growthbookDecryptionKey, rudderstackKey, growthbookOptions }: Options) => {
+    const initialise = ({
+        growthbookKey,
+        growthbookDecryptionKey,
+        rudderstackKey,
+        GBAttributes = {},
+        growthbookOptions,
+    }: Options) => {
         try {
             _rudderstack = RudderStack.getRudderStackInstance(rudderstackKey)
-            if (growthbookKey && growthbookDecryptionKey) {
+            if (Object.keys(GBAttributes).length > 0)
+                core_data = {
+                    ...core_data,
+                    ...(GBAttributes.geo_location !== undefined && { country: GBAttributes.country }),
+                    ...(GBAttributes.user_language !== undefined && { user_language: GBAttributes.user_language }),
+                    ...(GBAttributes.account_type !== undefined && { account_type: GBAttributes.account_type }),
+                    ...(GBAttributes.app_id !== undefined && { app_id: GBAttributes.app_id }),
+                    ...(GBAttributes.residence_country !== undefined && {
+                        residence_country: GBAttributes.residence_country,
+                    }),
+                    ...(GBAttributes.device_type !== undefined && { device_type: GBAttributes.device_type }),
+                    ...(GBAttributes.url !== undefined && { url: GBAttributes.url }),
+                }
+            if (!GBAttributes.id) GBAttributes.id = _rudderstack.getAnonymousId()
```

Comment:
what is the fallback for this?

Reply:
What do you mean by fallback?
In any case If we provide the GBAttributes.id then that will be used otherwise it will use the Anonymous id of RudderStack